### DISCUSSION TITLE
Is it possible to use Google for OAuth/OIDC?

### DISCUSSION THREAD

**Discussion by kcross88** [mealie-recipes/mealie#3477] [General]
Just updated and noticed that OIDC/OAuth is now an option for login/sign-ups which is awesome. I leverage Google's OAuth option for a few self-hosted services I share externally with family/friends, and would love to add it to Mealie. 

I set it up and notice its failing because it requires the "groups" scope, which from the documentation looks to be used in some optional settings to auto assign a group (or admin rights) to certain individuals. For my use case, I have these disabled as I only want a single group and I'm the only admin.

![image](https://github.com/mealie-recipes/mealie/assets/2367024/584fd9d5-14de-4e34-a5cb-d52b2026e5d9)

I tried adding it anyway and just not using the group features, but after manually adding the groups scope in the OAuth settings (assuming I have the right one), it appears to be considered a "sensitive scope" in Google. If I'm reading this right, it would require consent when signing in as its considered private user data, which I would like to avoid.

![image](https://github.com/mealie-recipes/mealie/assets/2367024/e5267681-00d9-4326-b0fa-c7714fb9467b)

Is there a way to use OAuth without that groups scope? Or does anyone have a working setup with Google they wouldn't mind sharing?

Apologies if this is a dumb question or in the incorrect category - still pretty new to OAuth. Not sure if that's even the only blocker to using Google.

Appreciate any help or guidance in advance!
KC

**Comment by cmintey** [comment]
Unfortunately, that `groups` scope wouldn't be the only barrier with Google. Mealie uses the OAuth Authorization Code flow with PKCE and no client_secret because of the Mealie architecture. Google doesn't seem to support this use case and will not give a token because the client_secret is missing. See [this Stack Overflow post](https://stackoverflow.com/questions/62507919/do-we-need-client-secret-when-using-pkce-in-openid-connect-authorization-code-fl)

**Reply by kcross88** [reply]
Ah, got it - really appreciate the response and the info on it. Might need to look into another provider like Authelia at some point.
